### ISSUE TITLE
Dataframe view: add an `Index` column for instances and support un/collapsing

### ISSUE THREAD

**Issue by abey79** [rerun-io/rerun#7066]
- Un/collapsing by clicking on the +
- Collapsing by clicking on the vertical line

<img width="151" alt="image" src="https://github.com/user-attachments/assets/d9a395c2-dae7-4dec-a8ad-4dac79567d40">

**Comment by abey79** [comment]
@gavrelina I like the look of the `[+]` / `[–]` thing, but shouldn't we be using the same triangle as everywhere else?

**Comment by gavrelina** [comment]
> @gavrelina I like the look of the `[+]` / `[–]` thing, but shouldn't we be using the same triangle as everywhere else?

You're right. 
Using ▶ and ▼ would make it more consistent. 

Initially [+] [-] was there to prevent repeating similar icons for 'sorting' and possible similarities with the 'static/temporal' components. But neither of those is used now (!! Except the sorting icon for [Chunk store browser](https://github.com/rerun-io/rerun/pull/7226), which we shall also change to <img width="64" alt="Screenshot 2024-08-26 at 14 33 46" src="https://github.com/user-attachments/assets/e1d376aa-2799-44a8-b8cb-ed03303bcefa">), hence it's not justified.


Screenshot with ▶ and ▼ for visibility: 
<img width="1165" alt="Screenshot 2024-08-26 at 14 14 39" src="https://github.com/user-attachments/assets/5107c17a-dda0-4c4e-8d74-d2aaf89eb2e9">

**Comment by gavrelina** [comment]
New proposal: 

Components are grouped by entities (before we can group them by archetypes through tagged components)
- Each of the instance could be "uncollapsed" until the last index of its entity. 
- One can uncollapse by pressing an icon, which appears only on hover
- Simultaneously the other instances are uncollapsed as well until the index of the initiated item. I.e. compA[0-5], compB[0-15], if I uncollapse via compA I will see CompA[0,1,2,3,4,5] and for CompB[0,1,2,3,4,5]+10 more. 
- If there are more instances for certain component we display "N more" with ability to uncollapse further. 
- When uncollapsed — each of the rows serves as a way to collapse everything back to a "summary row".
- When uncollapsed : we still see a summary row with the total number of instances for each component.
- If one scrolls down to uncollapsed instance batch that is longer then the rest -> we explore the sticky behavior for the latest known values for other components. 
- have a separate column with (shared) indices => to help users see which [N] is a certain value they are looking at instead of calculating (is it [61?] or [62?]).
<img width="1193" alt="Screenshot 2024-08-30 at 11 07 55" src="https://github.com/user-attachments/assets/4e6499e4-696d-4161-a36a-75818ea73081">

Interactions:
https://github.com/user-attachments/assets/2c6ef858-9aff-4181-ad96-d7fe15504adc

**Comment by abey79** [comment]
@gavrelina I'll do that in a follow-up PR, should be too back. Tracking this here:
- https://github.com/rerun-io/rerun/issues/7412
